backend/tournesol/lib/public_dataset.py

```diff
@@ -124,26 +124,50 @@ def get_individual_criteria_scores_data(poll_name: str) -> QuerySet:
     and return a non-evaluated Django `RawQuerySet`.
 
     An individual criteria score is a score computed by the algorithm for
-    a specific criterion, a specific entity and specific user:
+    a specific criterion, previously rated by a user for a specific entity.
 
         User X Entity X Computed criteria score
     """
     from tournesol.models import (  # pylint: disable=import-outside-toplevel
         ContributorRatingCriteriaScore,
     )
 
-    return (
-        ContributorRatingCriteriaScore
-        .objects
-        .select_related("contributor_rating")
-        .select_related("contributor_rating__poll")
-        .select_related("contributor_rating__user")
-        .select_related("contributor_rating__entity")
-        .filter(contributor_rating__poll__name=poll_name)
-        .filter(contributor_rating__is_public=True)
-        .order_by(
-            "contributor_rating__user__username",
-            "contributor_rating__entity__uid",
-            "criteria",
-        )
+    return ContributorRatingCriteriaScore.objects.raw(
+        """
+        SELECT
+            tournesol_contributorratingcriteriascore.id,
+            core_user.username,
+            tournesol_entity.uid,
+            tournesol_contributorratingcriteriascore.criteria,
+            tournesol_contributorratingcriteriascore.score,
+            tournesol_contributorratingcriteriascore.voting_right
+
+        FROM core_user
+
+        LEFT OUTER JOIN tournesol_contributorrating
```

Comment:
Out of curiosity, any idea why an outer join is preferred here?  
These row values are required to be present in order to match the following joins, so an inner join should be equivalent here, no?

Reply:
I'm going to double-check by running the query again. I might have instinctively used an outer join even if it was not required.

Reply:
I confirm the `OUTER JOIN` has no effect, it's a mistake.

Reply:
Fixed by https://github.com/tournesol-app/tournesol/pull/1265/commits/18b7d67ee688ba33926a38e0f7531470f101274a